acceptable/mocks.py

```diff
@@ -0,0 +1,461 @@
+        self._stop()
+
+
+def response_callback_factory(status=200, headers=None, body=None, json=None):
+    if headers is None:
+        headers = CaseInsensitiveDict()
+    else:
+        headers = CaseInsensitiveDict(headers)
+    if json is not None:
+        assert body is None
+        body = json_dumps(json).encode('utf-8')
+        if 'Content-Type' not in headers:
+            headers['Content-Type'] = 'application/json'
+    def response_callback(request):
+        return status, headers, body
+    return response_callback
+
+
+ok_no_content_response_callback = response_callback_factory()
+
+
+class Endpoint(object):
+    """Configurable endpoint.
+
+    Callable to create a context manager which activates and returns a mock
+    for this endpoint.
+    """
+    def __init__(self, base_url, service_name, endpoint_spec, response_callback=None):
+        self._url = urljoin(base_url, endpoint_spec.location)
+        self._service_name = service_name
+        self._name = endpoint_spec.name
+        self._methods = list(endpoint_spec.methods)
+        self._request_schema = endpoint_spec.request_schema
+        self._response_schema = endpoint_spec.response_schema
+        self._response_callback = response_callback
+    
+    @property
+    def service_name(self):
+        return self._service_name
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def url(self):
+        return self._url
+
+    @property
+    def methods(self):
+        return list(self._methods)
+
+    def disable_request_validation(self):
+        self._request_schema = None
+
+    def disable_response_validation(self):
+        self._response_schema = None
+
+    def disable_validation(self):
+        self.disable_request_validation()
+        self.disable_response_validation()
+
+    def set_request_schema(self, schema):
+        self._request_schema = schema
+
+    def set_response_schema(self, schema):
+        self._response_schema = schema
+
+    def set_response_callback(self, callback):
+        self._response_callback = callback
+        
+    def set_response(self, status=200, headers=None, body=None, json=None):
+        self._response_callback = response_callback_factory(status, headers, body, json)
+
+    def __call__(self, response_callback=None, call_recorder=None):
+        if call_recorder is None:
+            call_recorder = CallRecorder()
+        if response_callback is None:
+            response_callback = self._response_callback
+            if response_callback is None:
+                response_callback = ok_no_content_response_callback
+        return EndpointMockContextManager(
+            self._methods,
+            call_recorder,
+            self._service_name,
+            self._name,
+            self._url,
+            self._request_schema,
+            self._response_schema,
+            response_callback,
+        )
+
+
+class ServiceMock(object):
+    """User endpoints to endpoint mocks"""
```

Comment:
I don't really understand this docstring?

Reply:
Neither do I, will fix